src/yunohost/data_migrations/0009_dh_params.py

```diff
@@ -0,0 +1,30 @@
+import re
+import os
+
+from moulinette.utils.log import getActionLogger
+from moulinette.utils.filesystem import chown
+
+from yunohost.tools import Migration
+from yunohost.service import service_regen_conf
+
+command = "nice -n 19 openssl dhparam -out /etc/ssl/private/dh2048.pem -outform PEM -2 2048 -dsaparam 2> /var/log/yunohost/dhparam_generation.log && chown root:ssl-cert /etc/ssl/private/dh2048.pem && yunohost service regen-conf >> /var/log/yunohost/dhparam_generation.log && rm /etc/cron.hourly/yunohost-generate-dh-params\n"
+dhparams_file = "/etc/ssl/private/dh2048.pem"
+
+class MyMigration(Migration):
+    "This migration will add dh_params line and generate it in installed instance"
+
+    def migrate(self):
+
+        if os.path.exists(dhparams_file):
+            service_regen_conf(['nginx'])
```

Comment:
I still don't know what you expect to achieve by running this regen-conf :/ If the admin already generated a dhparam himself, either he modified the nginx conf already and that's not useful to run a regenconf here, or he didn't and the regen-conf which happen during the debian package upgrade will take care of it ...

Maybe we just wanna display a warning or info stating that the dhparam file already exists and there's nothing to do and that's it. Or just a return is fine as well I guess.

Reply:
mh... yes. Forgot there were a regen-conf during upgrade (I was alone with my migration in my head). ok.

Reply:
ok. Done.